code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -186,6 +187,7 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 		cached_gases[id][MOLES] -= removed_gases[id][MOLES]
 	garbage_collect()
 
+	SEND_SIGNAL(src, COMSIG_GASMIX_REMOVING)
```

Comment:
Any chance you could move this signals to the start of the proc?
In general it is preferable to make the events so that they may allow for an early return, in case of future need.
If your logic depends on them happening after the fact, then please change the names to something like `COMSIG_GASMIX_REMOVED` (same for the others) to make it clear this is after the event has already happened.

Reply:
Yes they do depend on the procs firing first. I will change it into the past tense then.